### ISSUE TITLE
Is there any way to reduce the scope to only specific domain requests? 

### ISSUE THREAD

**Issue by rdheepan** [jpillora/xdomain#181]
Say there are two applications which provides me the REST api's. 
1. appilction1.example.com
2. application2.example.com
They are deployed in two different servers. 

Out of those two, only application2.example.com is configured with xdomain and application1.example.com does not have any xdomain configurations. Currently the issues is, if I include xdomain.js in index.html, it affects all the xhr requests due to which the CORS request to application1.example.com is not working. 

Is there any way that I could use xdomain.js only for the requests which are made to application2.example.com?

**Comment by jpillora** [comment]
This should work if you only add application2.example.com as the slave, not
*.example.com

**Comment by rdheepan** [comment]
I hope I have the exact configuration as you mentioned above.My index.html has the following code block,

```
<script src="app/vendor/xdomain/xdomain.min.js"></script>
<script>
        xdomain.slaves({
                "https://application1.example.com": "/xdomain/Proxy.html"
        });
 </script>
```

But with this configuration, all my calls to https://application2.example.com and http://localhost:8080  are getting failed with a message like the following one,

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at
http://localhost:8080/rest/myapi. (Reason: CORS header 'Access-Control-Allow-Origin' missing).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at
http://localhost:8080/rest/myapi. (Reason: CORS request failed).
```

But both https://application2.example.com and localhost sends _Access-Control-Allow-Origin_ header in their responses.

**Comment by jpillora** [comment]
Do those errors go away when you comment out XDomain?

**Comment by rdheepan** [comment]
Yes. Without XDomain, localhost and other domain requests are working properly.

**Comment by jpillora** [comment]
Interesting, could you host it somewhere and send me link? or create a
redacted public version which reproduces it?

**Comment by rdheepan** [comment]
I'm afraid that I could host it in public. But I will try to make a redacted public version. BTW, I can see preflight OPTION request is sent for all localhost rest calls(which are getting failed with 404). But without XDomain, preflight requests were not sent. _[ I don't know whether this information is really helpful for you to identify the issue:)]_

**Comment by jpillora** [comment]
Ah yes, now I remember, I have seen this bug before and I couldn't figure
out why... Enabling XHook caused the OPTIONs requests not to be sent for
some reason. Don't worry about the demo, I'll think I've got my own demo of
this bug somewhere. I just need to find the time to debug it again...

**Comment by jpillora** [comment]
Created XHook issue here https://github.com/jpillora/xhook/issues/43

**Comment by ragabo** [comment]
What if I want to make the resources on slave be accessible to anonymous master.
For example
<script src="//unpkg.com/xdomain@0.8.2/dist/xdomain.min.js" master="*"></script>

Is this valid?

**Comment by jpillora** [comment]
Valid, but insecure. It's similar to adding CORS header
"access-control-allow-origin: *"
